### PR TITLE
update model functions and do_anomaly_detection

### PR DESCRIPTION
**Pull request by double-y**
### Description
Describe your fix here

### Checklist

Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [x] Pass devtools::test()
- [ ] Test installing from github
- [ ] Tested with Exploratory

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -1,13 +1,16 @@
 # Generated by roxygen2: do not edit by hand
 
 S3method(augment,multinom)
+S3method(augment,xgb.Booster)
+S3method(glance,xgb.Booster)
+S3method(tidy,xgb.Booster)
 export("%>%")
 export("%nin%")
+export()
```
Comment (hideaki) [inline NAMESPACE]: Is this necessary?

File: DESCRIPTION [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
+Version: 0.5.5.7
 Date: 2017-02-03
 Authors@R: c(person("Yosuke", "Yasuda", email = "double.y.919.quick@gmail.com", role = c("aut", "cre")))
 URL: https://github.com/exploratory-io/exploratory_func
 Description: Functions for Exploratory
 License: What license is it under?
 LazyData: TRUE
 Encoding: UTF-8
+Depends:
+    broom,
```
Comment (hideaki) [inline DESCRIPTION]: Do broom and urltools need to appear in both Depends and Imports?

File: NAMESPACE [hunk 3]
```diff
@@ -1,13 +1,16 @@
 # Generated by roxygen2: do not edit by hand
 
 S3method(augment,multinom)
+S3method(augment,xgb.Booster)
```
Comment (hideaki) [inline NAMESPACE]: Comment on what those S3method are for would help.

File: DESCRIPTION (hunk 2 above)
Comment (hideaki) [inline DESCRIPTION]: Please add comment on criteria on when a package should go to Depends of Imports.

File: R/broom_wrapper.R [hunk 4]
```diff
@@ -433,7 +433,12 @@ prediction_binary <- function(df, threshold = 0.5, ...){
   }
 
   # get actual value column name from model formula
-  actual_col <- all.vars(first_model$formula)[[1]]
+  actual_col <- if(!is.null(first_model$formula)) {
```
Comment (hideaki) [inline R/broom_wrapper.R]: I think we'd better check class of model.
I think $formula did not exist in lm etc. They have $terms instead.
Please take a look at what I did in model_stats().

File: R/build_xgboost.R [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+  if(is.integer(X) || is.logical(X)){
+    # X must be a double matrix
+    X <- matrix(as.double(X), ncol = ncol(X))
+  }
+
+  y <- data[[y_name]]
+
+  ret <- xgboost::xgboost(data = X, label = y, weight = weight, params = params, ...)
+  ret$fml <- formula
+  ret$X_names <- X_names
```
Comment (hideaki) [inline R/build_xgboost.R]: Why is X in uppercase while y is in lowercase around here?

File: R/build_xgboost.R [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+  data <- data %>% dplyr::select_if(function(col){is.numeric(col) || is.logical(col)})
+
+  if (any(X_names == '.')) {
+    X_names <- colnames(data)[names(data) != y_name]
+  }
+  for (X_name in X_names) {
+    data <- data[!is.na(data[[X_name]]), ]
+  }
+
+  # weight arguemtn should be evaluated using data name space
```
Comment (hideaki) [inline R/build_xgboost.R]: typo: arguemtn

File: R/importer.R [hunk 7]
```diff
@@ -23,3 +23,15 @@ urltools::url_decode
 anonymize <- function(vec, algo = "sha256", seed = 0, chars = letters, n_chars = 5L, ...){
   anonymizer::anonymize(vec, .algo = algo, .seed = seed, .chars = chars, .n_chars = n_chars, ...)
 }
+
+#' @importFrom tidytext tidy
+#' @export
+tidy
```
Comment (hideaki) [inline R/importer.R]: Doesn't these need qualifier "tidytext::" like other @importFrom entries?

File: R/build_xgboost.R [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+
+  ret <- xgboost::xgboost(data = X, label = y, weight = weight, params = params, ...)
+  ret$fml <- formula
+  ret$X_names <- X_names
+  ret$y_fct_level <- y_fct_level
+  ret$output_type <- output_type
+  ret
+}
+
+#' Glance xgb.Booster model
```
Comment (hideaki) [inline R/build_xgboost.R]: This is not Glance, is it?

File: R/build_xgboost.R [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+  .dots[["params"]] <- lazyeval::lazy(params)
+  .dots[["formula"]] <- lazyeval::lazy(formula)
+  build_model_(
+    data,
+    model_func = fml_xgboost,
+    nrounds = nrounds,
+    .dots = .dots)
+}
+
+#' Glance xgb.Booster model
```
Comment (hideaki) [inline R/build_xgboost.R]: This is not Glance, is it?

File: NAMESPACE (hunk 1 above)
Comment (double-y) [inline NAMESPACE]: Somehow, it's generated by devtools::check() and this file says "# Generated by roxygen2: do not edit by hand" so probably we shouldn't remove it

File: DESCRIPTION (hunk 2 above)
Comment (double-y) [inline DESCRIPTION]: Can't write a comment in DESCRIPTION file so I'll create Note section and write comment there

File: NAMESPACE (hunk 3 above)
Comment (double-y) [inline NAMESPACE]: Hmm, even if I write a comment here, devtools::check() removes it. Do you have an idea where to put comments?

File: R/broom_wrapper.R (hunk 4 above)
Comment (double-y) [inline R/broom_wrapper.R]: Okay, will try

File: R/build_xgboost.R (hunk 5 above)
Comment (double-y) [inline R/build_xgboost.R]: It was just so in original function from the url. I'll use lowercase

File: R/importer.R (hunk 7 above)
Comment (double-y) [inline R/importer.R]: Actually, I'm not really sure about behaviours around here but I needed this way of definition for tidy.xgb.Booster

File: R/build_xgboost.R (hunk 8 above)
Comment (double-y) [inline R/build_xgboost.R]: Right, will fix it

File: R/build_xgboost.R (hunk 9 above)
Comment (double-y) [inline R/build_xgboost.R]: Right, will fix it

File: R/build_xgboost.R (hunk 6 above)
Comment (double-y) [inline R/build_xgboost.R]: Will fix it

File: NAMESPACE (hunk 3 above)
Comment (hideaki) [inline NAMESPACE]: I see. It's ok then.

File: NAMESPACE (hunk 1 above)
Comment (hideaki) [inline NAMESPACE]: OK.

File: R/importer.R (hunk 7 above)
Comment (hideaki) [inline R/importer.R]: I see. Please leave that as a comment.

File: R/broom_wrapper.R (hunk 4 above)
Comment (double-y) [inline R/broom_wrapper.R]: This function is prediction_binary and this is supposed to be from glm or xgboost and what I need is left hand side, not terms. I'll add a validation for the class though

**Comment by double-y** [comment]
Test
- [x] test prediction after build_xgboost and build_lr in Exploratory
- [x] devtools::check()

File: R/broom_wrapper.R (hunk 4 above)
Comment (double-y) [inline R/broom_wrapper.R]: Added validation in the condition.

**Pull request merged**